### PR TITLE
FIX-#6443: Cast boolean columns before sum|mean|median groupby aggregations

### LINKED ISSUE #6443: HDK: sum|mean|median groupby aggregations fail for boolean columns
### Modin version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest released version of Modin.

- [X] I have confirmed this bug exists on the main branch of Modin. (In order to do this you can follow [this guide](https://modin.readthedocs.io/en/stable/getting_started/installation.html#installing-from-the-github-master-branch).)

### Reproducible Example

```python
df = pd.DataFrame({"A": [1, 2, 1], "B": [True, True, False]})
print(df.groupby("A").agg("sum"))
print(df.groupby("A").agg("mean"))
print(df.groupby("A").agg("median"))
```

### Issue Description

The code above fails with `AVG|SUM' is only valid on integer and floating point` error and the `median` aggregation is not valid. 

### Expected Behavior

```
   B
A   
1  1
2  1
     B
A     
1  0.5
2  1.0
     B
A     
1  0.5
2  1.0
```

### Error Logs

<details>

```python-traceback

Replace this line with the error backtrace (if applicable).

```

</details>

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

### PR DESCRIPTION
**Pull request by AndreyPavlenko**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6443 
- [x] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/experimental/core/storage_formats/hdk/query_compiler.py [hunk 1]
```diff
@@ -857,9 +857,9 @@ def dtypes(self):
         return self._modin_frame.dtypes
 
 
-_SUPPORTED_NUM_TYPE_CODES = set(np.typecodes["AllInteger"] + np.typecodes["Float"]) - {
-    np.dtype(np.float16).char
-}
+_SUPPORTED_NUM_TYPE_CODES = set(
+    np.typecodes["AllInteger"] + np.typecodes["Float"] + "?"
```
Comment (anmyachev) [inline modin/experimental/core/storage_formats/hdk/query_compiler.py:861]: Let's give name for `"?"` typecode.

**Review by anmyachev — DISMISSED** [review]
LGTM!

Please note on https://github.com/modin-project/modin/pull/6444/files#r1286338692

File: modin/experimental/core/storage_formats/hdk/query_compiler.py (hunk 1 above)
Comment (AndreyPavlenko) [reply modin/experimental/core/storage_formats/hdk/query_compiler.py:861]: I didn't find a dedicated constant for the boolean typecode. I can add an inline comment here. Is it ok?

Comment (anmyachev) [reply modin/experimental/core/storage_formats/hdk/query_compiler.py:861]: Yes

**Review by anmyachev — APPROVED** [review]

**Pull request merged**